scripts/release.sh

```diff
@@ -54,8 +54,16 @@ main() {
   fi
   RELEASE_VERSION="v${VERSION}"
   MINOR_VERSION=$(echo "${VERSION}" | cut -d. -f 1-2)
-  BRANCH=${BRANCH:-"release-${MINOR_VERSION}"}
-  REPOSITORY=${REPOSITORY:-"https://github.com/etcd-io/etcd.git"}
+
+
+  if [ "${CURRENT_BRANCH}" == 1 ]; then
+      # Trigger release in current branch, used in github workflow
+      REPOSITORY=$(pwd)
+      BRANCH=$(git branch)
+  else
+      REPOSITORY=${REPOSITORY:-"https://github.com/etcd-io/etcd.git"}
+      BRANCH=${BRANCH:-"release-${MINOR_VERSION}"}
```

Comment:
Maybe instead of introduction of new variable 'CURRENT_BRANCH' we can add support for 
`BRANCH="#CURRENT"` OR `BRANCH="."` magic value. This would be IMHO cleaner about interactions with other variables.

Reply:
It seems that the the approach `BRANCH="#CURRENT"` complicates the implementation, because we need to evaluate `[ "${BRANCH}" != "#CURRENT" ] ` in all places where `BRANCH` is used. Probably I do not get your point.  So I'd like to keep it as it's.

Reply:
So introducing a new variable `CURRENT_BRANCH` instead of updating the existing `BRANCH` is simpler.